stream-chat-android-core/src/main/java/io/getstream/chat/android/models/Poll.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2014-2024 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.models
+
+import java.util.Date
+
+/**
+ * The Poll object represents a poll in a channel.
+ *
+ * @property id The unique identifier of the poll.
+ * @property name The name of the poll.
+ * @property description The description of the poll.
+ * @property options The list of options for the poll.
+ * @property votingVisibility The visibility of the votes.
+ * If set to [VotingVisibility.ANONYMOUS], the votes will be anonymous.
+ * @property enforceUniqueVote If set to true, a user can only vote once.
+ * @property maxVotesAllowed The maximum number of votes a user can cast.
+ * @property allowUserSuggestedOptions If set to true, users can suggest new options.
+ * @property allowAnswers If set to true, users can vote.
+ * @property voteCountsByOption The number of votes for each option.
+ * @property votes The list of votes.
+ * @property ownVotes The list of votes cast by the current user.
+ */
+public data class Poll(
```

Comment:
We can use the `@Immutable` annotation for all these models to make them stable across multi-module architecture.

Reply:
Good point!!!